maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -569,6 +570,20 @@ else if ( cliRequest.commandLine.hasOption( CLIManager.BATCH_MODE )
         }
     }
 
+    /**
+     * Disables the colour output in the case that the {@link CLIManager#BATCH_MODE} option
+     * or {@link CLIManager#LOG_FILE} option was given (or both). In those cases, ANSI output is never feasible.
+     * @param cliRequest the arguments as request pojo.
+     */
+    private void disableColorsInLogfileOrBatch( CliRequest cliRequest )
```

Comment:
I'd prefer a more general method name: applyColorMode. This way we can add more logic when necessary